tests/test_enterprise/views/test_enterprise_selection.py

```diff
@@ -0,0 +1,165 @@
+# -*- coding: utf-8 -*-
+"""
+Tests for the ``EnterpriseSelectionView`` view of the Enterprise app.
+"""
+
+from __future__ import absolute_import, unicode_literals
+
+import os
+import tempfile
+
+import ddt
+import mock
+from pytest import mark
+
+from django.conf import settings
+from django.core.urlresolvers import reverse
+from django.test import Client, TestCase
+
+from enterprise.forms import ENTERPRISE_SELECT_SUBTITLE
+from enterprise.models import EnterpriseCustomerUser
+from test_utils.factories import EnterpriseCustomerFactory, EnterpriseCustomerUserFactory, UserFactory
+
+
+@mark.django_db
+@ddt.ddt
+class TestEnterpriseSelectionView(TestCase):
+    """
+    Test EnterpriseSelectionView.
+    """
+    url = reverse('enterprise_select_active')
+
+    def setUp(self):
+        self.user = UserFactory.create(is_active=True)
+        self.user.set_password("QWERTY")
+        self.user.save()
+        self.client = Client()
+        super(TestEnterpriseSelectionView, self).setUp()
+
+        self.success_url = 'http://localhost:8000/enterprise/grant_data_sharing_permissions'
+        enterprises = ['Gryffindor', 'Hufflepuff', 'Ravenclaw', 'Slytherin']
+        for enterprise in enterprises:
+            enterprise_customer = EnterpriseCustomerFactory(name=enterprise)
+            EnterpriseCustomerUserFactory(
+                user_id=self.user.id,
+                enterprise_customer=enterprise_customer
+            )
+
+        enterprises = EnterpriseCustomerUser.objects.filter(
+            user_id=self.user.id
+        ).values_list(
+            'enterprise_customer__uuid', 'enterprise_customer__name'
+        )
+        self.enterprise_choices = [(str(k), v) for k, v in enterprises]
+
+        # create a temporary template file
+        # rendering `enterprise/enterprise_customer_select_form.html` fails becuase of dependency on edx-platform
+        tpl = tempfile.NamedTemporaryFile(
```

Comment:
Creating a temp file does not make sense. Can we mock the the template?

Reply:
I tried but didn't found a way to mock the template. `ForView.template_name` must be a path to a template that is why we need a temporary file.